identity/aziot-identityd/src/identity.rs

```diff
@@ -719,6 +719,14 @@ impl IdentityManager {
         assert!(!status.eq_ignore_ascii_case("assigning"));
 
         let mut state = operation.registration_state.ok_or(Error::DeviceNotFound)?;
+
+        if state.status == Some("failed".to_string()) {
```

Comment:
`state.status.as_deref() == Some("failed")`, to avoid allocating a String just for the comparison.

Reply:
Fixed